src/app/Common.DB.Postgres/Services/UsersService.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Common.DB.Postgres.DAL.Documents;
+using Common.DB.Postgres.DAL.Interfaces;
+using Common.Models;
+using Common.Services;
+using Common.Services.EmailService;
+using Common.Services.UserService;
+using Common.Utils;
+using LinqToDB;
+
+namespace Common.DB.Postgres.Services
+{
+    public class UserService : BaseDocumentService<User>, IUserService
+    {
+        private readonly IUserRepository _userRepository;
+        private readonly IEmailService _emailService;
+
+        public UserService(IUserRepository userRepository,
+            IEmailService emailService) : base(userRepository)
+        {
+            _userRepository = userRepository;
+            _emailService = emailService;
+        }
+
+        public async Task<IUser?> FindByEmailAsync(string email)
+        {
+            return await _userRepository.GetQuery().FirstOrDefaultAsync(x => x.Email == email);
```

Comment:
can we specify ignore case here?

Reply:
Do you mean to add a "string userIdToExclude" parameter?

Reply:
I mean compare emails ignoring case, because as I know test@gmail.com and TEST@gmail.com are the same emails